### PR TITLE
Further hardened delta

### PR DESCRIPTION
**Pull request by DavidM-D**
We use re-randomization to mitigate the issues described in Groth,Shoup 22 (even though our model deviates substantially from theirs).

The properties of a good delta is it has to be unpredictable before the signature request is created and it can only be used once for a signing protocol.

Since we current use `H ( VRF, H(signature_request, ...) )` we guarantee the first point, but there are situations where we may try to sign again using the same delta. e.g. if our protocol crashes or someone manages to get us to ingress a duplicate message. By including the public R paramater of the presignature and since we guarantee that we don't reuse presignatures, this PR guarantees the second property required.

To read more we have a deeper dive into the subject here https://docs.google.com/document/d/1-Ibv5R5mokSlcV1FhKSGmAK6jHqyRwmffe2moTzR_k4/edit.

### CODE REVIEW

**Comment by DavidM-D** [comment]
We'll wait for Alessandra to come back before we merge this.

**Review by ChaoticTempest — COMMENTED** [review]
This is good to me because even if we retry signature generation on a failed one, a new presignature will always get used so we are able to guarantee second property.

But does changing this change any of the derived keys? Or only epsilon is the value that changes the keys?

**Comment by DavidM-D** [comment]
Nope derived keys are only dependent on epsilon, and the hard coded signature derivation tests check that this is true.

**Review by ChaoticTempest — APPROVED** [review]

**Pull request merged**
